Expose capybara client to custom loaders

while1malloc0 opened the pull request:
Implements the feature requested in https://github.com/percy/percy-capybara/issues/5

fotinakis commented on spec/lib/percy/capybara_spec.rb:
```diff
@@ -76,4 +76,14 @@
+    it 'sets the current capybara clients custom loader flag' do
+      capybara_client = Percy::Capybara::Client
+      expect(Percy::Capybara.capybara_client.custom_loader?).to eq(false)
+      Percy::Capybara.use_loader DummyLoader
```
For style consistency with other things around, use parens here `use_loader(DummyLoader)`

fotinakis commented on lib/percy/capybara/client.rb:
```diff
@@ -42,8 +43,12 @@ def disable!
         @enabled = false
       end
 
+      def custom_loader?
```
Drop this method to simplify things, just use `if custom_loader` as the check below and `expect(Percy::Capybara.capybara_client.custom_loader).to be` or `.not_to be` in the tests.

fotinakis commented on lib/percy/capybara/client.rb:
```diff
@@ -62,7 +67,10 @@ def failed?
-        if sprockets_environment && sprockets_options
+        if custom_loader?
+          Percy.logger.debug { 'Using a custom loader to discover assets.' }
+          @custom_loader.new(options)
```
Use `custom_loader` here instead of `@custom_loader` (just a personal style preference to prefer using accessors that will raise errors if spelled wrong instead of instance vars that will silently return `nil`).

fotinakis commented:
Nice, I like this! Some comments above, back to you @while1malloc0

while1malloc0 commented:
I think I addressed all of the comments. I'm not 100% sure that the spec on `Percy::Capybara.use_loader` is testing the right way, but that might just be my aversion to checking whether or not something is nil in specs.

fotinakis commented on spec/lib/percy/capybara_spec.rb:
```diff
@@ -76,4 +76,14 @@
+
+    it 'sets the current capybara clients custom loader' do
+      capybara_client = Percy::Capybara::Client
+      expect(Percy::Capybara.capybara_client.custom_loader).to be nil
```
I don't usually pass anything into `be`, just use it as a check for existence or not. The normal form is to to use `.to_not be` or the more explicit `.to be_nil` here, and then `.to be` or the more explicit `.to_not be_nil` below, but I prefer the shorter form.

fotinakis commented:
LGTM, will merge and tag.

fotinakis commented:
Released as v2.2.0 of the gem — thanks for contributing this!
